sources/packages/backend/apps/api/src/route-controllers/restriction/_tests_/e2e/restriction.institution.controller.getStudentRestrictionDetail.e2e-spec.ts

```diff
@@ -0,0 +1,154 @@
+  BEARER_AUTH_TYPE,
+  createTestingAppModule,
+  getAuthRelatedEntities,
+  getInstitutionToken,
+  InstitutionTokenTypes,
+} from "../../../../testHelpers";
+import * as request from "supertest";
+import { getUserFullName } from "../../../../utilities";
+
+describe("RestrictionInstitutionController(e2e)-getStudentRestrictions", () => {
+  let app: INestApplication;
+  let appDataSource: DataSource;
+  let collegeF: Institution;
+  let collegeC: Institution;
+  let collegeFLocation: InstitutionLocation;
+  let collegeCLocation: InstitutionLocation;
+  let applicationRepo: Repository<Application>;
+
+  beforeAll(async () => {
+    const { nestApplication, dataSource } = await createTestingAppModule();
+    app = nestApplication;
+    appDataSource = dataSource;
+
+    const { institution } = await getAuthRelatedEntities(
+      appDataSource,
+      InstitutionTokenTypes.CollegeFUser,
+    );
+    collegeF = institution;
+    collegeFLocation = createFakeInstitutionLocation(collegeF);
+    await authorizeUserTokenForLocation(
+      appDataSource,
+      InstitutionTokenTypes.CollegeFUser,
+      collegeFLocation,
+    );
+    applicationRepo = appDataSource.getRepository(Application);
+  });
+
+  it("Should return forbidden 403 error code when a non BC Public Institution User tries to get student restriction by student restriction id.", async () => {
+    // Arrange
+    const { institution } = await getAuthRelatedEntities(
+      appDataSource,
+      InstitutionTokenTypes.CollegeCUser,
+    );
+    collegeC = institution;
+    collegeCLocation = createFakeInstitutionLocation(collegeC);
+    await authorizeUserTokenForLocation(
+      appDataSource,
+      InstitutionTokenTypes.CollegeCUser,
+      collegeCLocation,
+    );
+    const student = await saveFakeStudent(appDataSource);
+    const application = createFakeApplication({
+      location: collegeCLocation,
+      student,
+    });
+    await applicationRepo.save(application);
+    const studentRestriction = await saveFakeStudentRestriction(
+      appDataSource,
+      { student, application, creator: student.user },
+      {
+        restrictionCategory: "Verification",
+        notificationType: RestrictionNotificationType.NoEffect,
+      },
+    );
+    const institutionUserToken = await getInstitutionToken(
+      InstitutionTokenTypes.CollegeCUser,
+    );
+    const endpoint = `/institutions/restriction/student/${student.id}/studentRestriction/${studentRestriction.id}`;
+    // Act/Assert
+    await request(app.getHttpServer())
+      .get(endpoint)
+      .auth(institutionUserToken, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.FORBIDDEN);
+  });
+
+  it("Should return forbidden 403 error code when a BC Public Institution User tries to get student restriction for a student which they don't have access to.", async () => {
+    // Arrange
+    const student = await saveFakeStudent(appDataSource);
+    const studentRestriction = await saveFakeStudentRestriction(
```

Comment:
Do we need the restriction data to be created for this scenario?

Reply:
Yes, I created a `studentRestriction` because the endpoint to be tested requires a studentRestriction id.

Reply:
@sh16011993 the point from @dheepak-aot was, if the validation happens before the controller reaches the restriction query, why do we need to create restriction data?